front-end/src/components/gallery/Gallery.js

```diff
@@ -77,24 +95,71 @@ const PagButton = styled.button`
 
 const Gallery = () => {
     const dispatch = useDispatch();
-    const products = useSelector(selectProduct); 
-    const filters = useSelector(selectFilters); 
+    const { search } = useLocation();
+    const searchParams = new URLSearchParams(search);
+
+    const history = useHistory();
+    const products = useSelector(selectProduct);
+    const filters = useSelector(selectFilters);
     const [currPage, setCurrPage] = useState(1);
-    const [categoryOption, setCategoryOption] = useState({});
+    const [categoryOption, setCategoryOption] = useState('');
     const [sizeOption, setSizeOption] = useState([]);
     const [colorOption, setColorOption] = useState([]);
+    const [priceOption, setPriceOption] = useState('');
+
 
     useEffect(() => {
-        productService.getProducts(dispatch, currPage, categoryOption, sizeOption, colorOption);
+        if (searchParams.get('category')) {
+            setCategoryOption(searchParams.get('category'))
+        }
+        if (searchParams.get('size')) {
+            setSizeOption((sizeOption) => [...sizeOption, searchParams.get('size')]);
+        }
+        if (searchParams.get('color')) {
+            setColorOption((colorOption) => [...colorOption, searchParams.get('color')]);
+        }
+        if (searchParams.get('prise')) {
+            setPriceOption(searchParams.get('prise'))
 
-    }, [dispatch, currPage, categoryOption, sizeOption, colorOption]);
+        }
+    }, [])
+
+    useEffect(() => {
+        let categoryUrl = '';
+        let sizeUrl = '';
+        let colorUrl = '';
+        let priceUrl = '';
+        if (categoryOption) {
+            categoryUrl = utilFunc.stringifyUrl('category', categoryOption);
+        }
+        if (sizeOption.length > 0) {
+            sizeUrl = utilFunc.stringifyUrl('size', sizeOption);
+        }
+        if (colorOption.length > 0) {
+            colorUrl = utilFunc.stringifyUrl('color', colorOption);
+        }
+        if (priceOption) {
+            priceUrl = utilFunc.stringifyUrl('price', priceOption);
+        }
+        console.log(colorUrl)
+
+        history.push({
```

Comment:
Should this call be in this useEffect? Isn't it better to create another useEffect with [] dependancies and it will be fired only the first time the page is rendered? What do you think?

Reply:
This useEffect is for building the URL when some prop is added. I have another that is fired only on the first render to set the state if there is anything to set in the URL. I am not sure if I need to separate it because it is used to build the URL and make the call to the backend with the same data.